packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -587,6 +607,13 @@ - (FlutterError *_Nullable)onListenWithArguments:(id _Nullable)arguments
   // This line ensures the 'initialized' event is sent when the event
   // 'AVPlayerItemStatusReadyToPlay' fires before _eventSink is set (this function
   // onListenWithArguments is called)
+  // and also send error in similar case with 'AVPlayerItemStatusFailed'
+  // https://github.com/flutter/flutter/issues/151475
+  // https://github.com/flutter/flutter/issues/147707
+  if (self.player.currentItem.status == AVPlayerItemStatusFailed) {
```

Comment:
So this is the part that actually fix the bug? Can you add a comment explaining how we ended up with Failed state here?

Reply:
Sorry I did not manage to proceed with this. Yes when `AVPlayerItemStatusFailed` fires before `_eventSink` is set. `AVPlayerItem` is "started" to do its things and can asynchronously become `ReadyToPlay` or `Failed` before the dart side starts to listen where `onListenWithArguments` is called. My comment here follows up that comment before which tries to explain this but that previous fix connected with that comment omitted `Failed` state as possible state here which this line fixes (if player item can end up with success here it can also end up with failure, so that previous comment should explain it).